src/atomate2/vasp/flows/ferroelectric.py

```diff
@@ -0,0 +1,123 @@
+    interpolate_structures,
+    polarization_analysis,
+)
+
+__all__ = ["FerroelectricMaker"]
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class FerroelectricMaker(Maker):
+    """
+    Maker to calculate polarization of a polar material.
+
+    Parameters
+    ----------
+    name : str
+        Name of the flows produced by this maker.
+    nimages: int
+        Number of interpolations calculated from polar to nonpolar structures,
+        including the nonpolar.
+    """
+
+    name: str = "ferroelectric"
+    nimages: int = 9
+    symprec: float = SETTINGS.SYMPREC
+    relax: bool | tuple = False
+    bulk_relax_maker: BaseVaspMaker | None = field(
+        default_factory=lambda: DoubleRelaxMaker.from_relax_maker(RelaxMaker())
+    )
+    lcalcpol_maker: BaseVaspMaker = field(default_factory=PolarizationMaker)
+
+    def make(
+        self,
+        polar_structure: Structure,
+        nonpolar_structure: Structure,
+        prev_vasp_dir: str | Path | None = None,
+    ):
+        """
+        Make flow to calculate the polarization.
+
+        Parameters
+        ----------
+        polar_structure : .Structure
+            A pymatgen structure of the polar phase.
+        nonpolar_structure : .Structure
+            A pymatgen structure of the nonpolar phase.
+        prev_vasp_dir : str or Path or None
+            A previous vasp calculation directory to use for copying outputs.
+        """
+        jobs = []
+        prev_vasp_dir_p, prev_vasp_dir_np = None, None
+
+        if isinstance(self.relax, bool):
+            self.relax = (self.relax, self.relax)
+
+        if self.relax[0]:
+            # optionally relax the polar structure
+            relax_p = self.bulk_relax_maker.make(polar_structure)
+            relax_p.append_name(" polar")
+            jobs.append(relax_p)
+            polar_structure = relax_p.output.structure
+            prev_vasp_dir_p = relax_p.output.dir_name
+
+        logger.info(f"{type(polar_structure)}")
+
+        polar_lcalcpol = self.lcalcpol_maker.make(
+            polar_structure, prev_vasp_dir=prev_vasp_dir_p
+        )
+        polar_lcalcpol.name += " polar"
+        jobs.append(polar_lcalcpol)
+        polar_structure = polar_lcalcpol.output.structure
+
+        if self.relax[1]:
+            # optionally relax the nonpolar structure
+            relax_np = self.bulk_relax_maker.make(nonpolar_structure)
+            relax_np.append_name(" nonpolar")
+            jobs.append(relax_np)
+            nonpolar_structure = relax_np.output.structure
+            prev_vasp_dir_np = relax_np.output.dir_name
+
+        nonpolar_lcalcpol = self.lcalcpol_maker.make(
+            nonpolar_structure, prev_vasp_dir=prev_vasp_dir_np
+        )
+        nonpolar_lcalcpol.append_name(" nonpolar")
+        jobs.append(nonpolar_lcalcpol)
+        nonpolar_structure = nonpolar_lcalcpol.output.structure
+
+        interp_lcalcpol = interpolate_structures(
```

Comment:
It could be clearer to break this into two functions:
1. To generate the interpolated structures.
2. To do the replace and submit the calculations.

The second job should probably have a `prev_vasp_dir` option (to enable `auto_ispin` to work) and also pass in the polarisation maker. Currently it just generates a new maker in the function itself which isn't customisable.